spec/unit/application/apply_spec.rb

```diff
@@ -484,6 +484,37 @@ def temporary_catalog(content = '"something"')
     end
   end
 
+  describe "when really executing" do
+    include PuppetSpec::Files
+    let(:testfile) { tmpfile('secret_file_name') }
+    let(:resourcefile) { tmpfile('resourcefile') }
+    let(:classfile) { tmpfile('classfile') }
+
+    it "should not expose sensitive data in the relationship file" do
+      @apply.options[:code] = <<-CODE
+        $secret = Sensitive('cat #{testfile}')
+
+        exec { 'do it':
+          command => $secret,
+          path    => '/bin/'
+        }
+      CODE
+
+      @apply.options[:write_catalog_summary] = true
+
+      Puppet.settings[:resourcefile] = resourcefile
+      Puppet.settings[:classfile] = classfile
+
+      expect { @apply.main }.to exit_with 0
+
+      fh = open resourcefile
+      result = fh.read
+      fh.close
+
+      expect(result).not_to match(/secret_file_name/)
```

Comment:
I'd also add an expectation for what we do expect to find so that if `result` is empty the test doesn't silently pass

Reply:
Added, thanks.